zproject/backends.py

```diff
@@ -414,7 +414,7 @@ class ZulipLDAPAuthBackend(ZulipLDAPAuthBackendBase):
     def authenticate(self, username, password, realm_subdomain=None, return_data=None):
         # type: (Text, str, Optional[Text], Optional[Dict[str, Any]]) -> Optional[UserProfile]
         try:
-            if settings.REALMS_HAVE_SUBDOMAINS:
+            if settings.REALMS_HAVE_SUBDOMAINS and realm_subdomain is not None:
```

Comment:
This is a code path that's likely to be significantly restructured this week, so might not be worth working on right now.